tools/update-engine-size/src/main.rs

```diff
@@ -0,0 +1,76 @@
+use std::{
+    fs::{self, OpenOptions},
+    path::Path,
+};
+
+use clap::Parser;
+use color_eyre::eyre::eyre;
+use serde::Serialize;
+
+/// This tool updates the size of the engine binaries in the CSV file
+/// on CI for tracking their changes over time.
+#[derive(Parser)]
+struct Args {
+    /// Path to the CSV database
+    #[arg(long)]
+    db: String,
+
+    /// Current git branch
+    #[arg(long)]
+    branch: String,
+
+    /// Current git commit
+    #[arg(long)]
+    commit: String,
+
+    /// List of engine files
+    files: Vec<String>,
+}
+
+#[derive(Serialize)]
+struct DbEntry<'a> {
+    date_time: &'a str,
+    branch: &'a str,
+    commit: &'a str,
+    file: &'a str,
+    size_bytes: u64,
+}
+
+fn main() -> color_eyre::Result<()> {
```

Comment:
probably remove it to save on compile time, `Box<dyn Error>` is more than enough for a simple program like this